### ISSUE TITLE
Improve analytics events

### ISSUE THREAD

**Issue by jviotti** [resin-io/etcher#1100]
For debugging and UX design purposes (e.g: to make effective design decisions based on real user behaviour). These are the events we have right now:

**For our users:** you can always disable error reporting and analytics from the settings page (its optional), but we should probably make that more obvious (something to discuss in another issue)

- Application start
- Checking for updates
- Notifying update
- Select image
- Re-select image
- Select Drive
- Reselect drive
- Flash
- Restart
- Restart after failure
- Done
- Validation Error
- Out of Space
- Flash Error
- Invalid Image

These are the events we want to have:

### Global

- [x] Application Start
- [x] Application attempted to close while flashing
- [x] Application attempted to close while not flashing
- [x] Show exit while flashing warning
- [x] Cancel flashing warning
- [x] Agree flashing warning

### Updates

- [x] Checking for Updates (include current version)
- [x] Notifying Update (include the latest version)
- [x] Update accepted
- [x] Update dismissed
- [x] Update dismissed and put to sleep
- [x] No Update

### Image

- [x] Open image selector
- [x] Image selector dismissed
- [x] Select image (include image data)
- [x] Change image
- [x] Show image info modal
- [x] Show image info tooltip
- [x] Multiple images inside archive

### Drive

- [ ] No drives (only after you select the image)
- [x] Auto-select drive (include drive)
- [ ] Auto-deselect drive (include drive and image)
- [x] Select drive (include drive, and if using unsafe mode)
- [x] Open drive modal
- [x] Drive modal dismissed
- [x] Toggle drive selection (include drive)

### Header

- [ ] Click help icon
- [ ] Click on settings

### Footer

- [x] Click on Etcher logo
- [x] Click on “an open source project by”
- [x] Click on Resin.io logo
- [x] Click version

### Settings

- [x] Click on back button
- [x] Toggle setting (include setting name and new value)
- [x] Accept setting warning (include setting name)
- [x] Dismiss setting warning (include setting name)

### Flash

- [ ] Click flash (include drive and image, is validation enabled, is unmounting enabled)
- [ ] Show elevation dialog
- [ ] No elevation needed
- [ ] Elevation successful
- [ ] Elevation rejected
- [ ] Flashing process started
- [ ] Flashing half through
- [ ] Flashing is about to complete
- [ ] Flashing error
- [ ] Validation started
- [ ] Validation half through
- [ ] Validation is about to complete
- [ ] Validation failed
- [ ] Out of Space error
- [ ] Retry after error
- [ ] Dismiss error modal
- [ ] Unmounting before flash (include drives that are getting unmounted)
- [ ] Unmounting after flash (include drives that are getting unmounted)
- [ ] Unmount successful (before and after flash)
- [ ] Unmount error (before and after flash)
- [ ] Flash complete (include total elapsed time)
- [ ] Use same image
- [ ] Use different image

Notice that for all modal events, we should include whether the modal was dismissed by clicking the "X" icon, or by clicking somewhere in the app.

We want to report the following user information:

- Operating system
- Operating system version
- Operating system architecture
- Etcher architecture
- Total RAM of the computer
- CPU information (number of cores, etc)

**Comment by lurch** [comment]
If it helps, this is what `grep -R analytics.logEvent` reports on the current codebase:
```
lib/gui/app/components/modal/services/modal.js:    analytics.logEvent('Open modal', {
lib/gui/app/components/modal/services/modal.js:          analytics.logEvent('Modal accepted', {
lib/gui/app/components/modal/services/modal.js:            analytics.logEvent('Modal rejected', {
lib/gui/app/components/modal/services/modal.js:          analytics.logEvent('Modal rejected', {
lib/gui/app/components/safe-webview.js:      analytics.logEvent(event)
lib/gui/app/components/safe-webview.js:      analytics.logEvent(message.data || message)
lib/gui/app/components/drive-selector/controllers/drive-selector.js:        analytics.logEvent('Toggle drive', {
lib/gui/app/components/drive-selector/controllers/drive-selector.js:        analytics.logEvent('Drive selected (double click)')
lib/gui/app/components/update-notifier.js:    analytics.logEvent('Close update modal', {
lib/gui/app/components/flash-error-modal/services/flash-error-modal.js:        analytics.logEvent('Restart after failure')
lib/gui/app/modules/analytics.js: * analytics.logEvent('Select image', {
lib/gui/app/modules/image-writer.js:  analytics.logEvent('Flash', analyticsData)
lib/gui/app/modules/image-writer.js:      analytics.logEvent('Elevation cancelled', analyticsData)
lib/gui/app/modules/image-writer.js:      analytics.logEvent('Done', analyticsData)
lib/gui/app/modules/image-writer.js:      analytics.logEvent('Validation error', analyticsData)
lib/gui/app/modules/image-writer.js:      analytics.logEvent('Drive unplugged', analyticsData)
lib/gui/app/modules/image-writer.js:      analytics.logEvent('Input/output error', analyticsData)
lib/gui/app/modules/image-writer.js:      analytics.logEvent('Out of space', analyticsData)
lib/gui/app/modules/image-writer.js:      analytics.logEvent('Child died unexpectedly', analyticsData)
lib/gui/app/modules/image-writer.js:      analytics.logEvent('Flash error', _.merge({
lib/gui/app/app.js:  analytics.logEvent('Application start', {
lib/gui/app/app.js:    analytics.logEvent('Not checking for updates', {
lib/gui/app/app.js:  analytics.logEvent('Checking for updates', {
lib/gui/app/app.js:      analytics.logEvent('Update notification skipped', {
lib/gui/app/app.js:      analytics.logEvent('Update notification skipped', {
lib/gui/app/app.js:    analytics.logEvent('Notifying update', {
lib/gui/app/app.js:    analytics.logEvent('Update check user error', {
lib/gui/app/app.js:      analytics.logEvent('Close application', {
lib/gui/app/app.js:    analytics.logEvent('Close attempt while flashing')
lib/gui/app/app.js:        analytics.logEvent('Close confirmed while flashing', {
lib/gui/app/app.js:      analytics.logEvent('Close rejected while flashing')
lib/gui/app/app.js:    analytics.logEvent('Navigate', {
lib/gui/app/os/open-external/services/open-external.js:    analytics.logEvent('Open external link', {
lib/gui/app/pages/main/controllers/image-selection.js:      analytics.logEvent('Invalid image', image)
lib/gui/app/pages/main/controllers/image-selection.js:        analytics.logEvent('Possibly Windows image', image)
lib/gui/app/pages/main/controllers/image-selection.js:        analytics.logEvent('Missing partition table', image)
lib/gui/app/pages/main/controllers/image-selection.js:      return analytics.logEvent('Select image', image)
lib/gui/app/pages/main/controllers/image-selection.js:    analytics.logEvent('Open image selector')
lib/gui/app/pages/main/controllers/image-selection.js:        analytics.logEvent('Image selector closed')
lib/gui/app/pages/main/controllers/image-selection.js:    analytics.logEvent('Reselect image', {
lib/gui/app/pages/main/controllers/drive-selection.js:      analytics.logEvent('Select drive', {
lib/gui/app/pages/main/controllers/drive-selection.js:    analytics.logEvent('Reselect drive')
lib/gui/app/pages/main/controllers/main.js:    analytics.logEvent('Show selected image tooltip', {
lib/gui/app/pages/finish/controllers/finish.js:    analytics.logEvent('Restart', options)
lib/gui/app/pages/settings/controllers/settings.js:    analytics.logEvent('Toggle setting', {
```
